client/my-sites/checkout/composite-checkout/use-prepare-product-for-cart.js

```diff
@@ -210,28 +209,39 @@ function useAddProductFromSlug( {
 			debug( 'waiting on products/plans fetch' );
 			return;
 		}
-		if ( ! product ) {
-			debug( 'there is a request to add a product but no product was found', productAlias );
+		if ( ! validProducts || validProducts.length < 1 ) {
+			debug(
+				'there is a request to add one ore more products but no product was found',
+				productAliasFromUrl,
+				validProducts
+			);
 			setState( { canInitializeCart: true } );
 			return;
 		}
-		const cartProduct = createItemToAddToCart( {
-			productAlias,
-			product_id: product.product_id,
-			isJetpackNotAtomic,
-			isPrivate,
-		} );
-		if ( ! cartProduct ) {
-			debug( 'there is a request to add a product but creating an item failed', productAlias );
+
+		const cartProducts = validProducts.map( ( product ) =>
+			createItemToAddToCart( {
+				productAlias: product.product_alias,
+				product_id: product.product_id,
+				isJetpackNotAtomic,
+				isPrivate,
+			} )
+		);
+
+		if ( ! cartProducts || cartProducts.length < 1 ) {
```

Comment:
I think `cartProducts` will always be an array here so we could remove the first condition.